src/main/java/qna/domain/Question.java

```diff
@@ -44,7 +48,26 @@ public boolean isOwner(User writer) {
         return this.writer.equals(writer);
     }
 
+    private void checkOwner(User writer) {
+        if (!isOwner(writer)) {
+            throw new RuntimeException("질문을 삭제할 권한이 없습니다.");
+        }
+    }
+
+    public DeleteHistories delete(User writer) {
+        DeleteHistories histories = new DeleteHistories();
+        Answers answers = this.answers.getNotDeleted();
+
+        checkOwner(writer);
+        answers.checkIsOwner(writer);
+
+        this.deleted = true;
+
+        return histories.delete(answers, DeleteHistory.question(id, writer));
```

Comment:
삭제 기록들 객체에게 삭제하라는 책임은 조금 어색해보이기도 하는데요 🤔 
아래 처럼 같이 개선해볼 수도 있을 것 같아요. 참고만 해주세요 :)

```suggestion
        return DeleteHistories.of(answers.delete, DeleteHistory.question(id, writer));
```

Reply:
네 뭔가 도메인의 역할에 따라 작성되기 보다는 클래스의 변수에 위임하면서 만든게 아닌가 싶네요 감사합니다